src/shared/hooks/useInfinityData.ts

```diff
@@ -0,0 +1,47 @@
+import { useAppDispatch } from 'redux/store';
+import { AsyncThunk } from '@reduxjs/toolkit';
+import { AxiosResponse } from 'axios';
+
+import { getInfinityScrollData } from 'shared/utils/getInfinityScrollData';
+import { useIntersectionObserver } from 'shared/hooks';
+
+export type UseInfinityDataType = {
+  rootSelector: string;
+  targetSelector: string;
+  listSize: number;
+  totalSize: number;
+  isLoading: boolean;
+  limitPerPage: number;
+  getListThunk: AsyncThunk<AxiosResponse, any, any>;
+};
+export const useInfinityData = ({
+  rootSelector,
+  targetSelector,
+  totalSize,
+  listSize,
+  isLoading,
+  limitPerPage,
```

Comment:
```suggestion
  limitPerPage = DEFAULT_ROWS_PER_PAGE,
```
and then it is not needed to pass this parameter when using this hook if the limit per page is the default